packages/graphql-explorer/src/logic/QueryBuilder.ts

```diff
@@ -134,33 +138,61 @@ export default class QueryBuilder {
     }
     if (isArray(input)) {
       const arrayItems = input
-        .map((i) => this.serializeInputValue(i))
+        .map((i) => this.serializeInputValue(i, rawArgType))
         .join(', ');
       return `[${arrayItems}]`;
     }
     if (isPlainObject(input)) {
+      const objectFields = g.assertInputObjectType(rawArgType).getFields();
+
       const objectItems = Object.entries(input)
         .filter(([, v]) => v !== undefined)
-        .map(([k, v]) => `${k}: ${this.serializeInputValue(v)}`)
+        .map(
+          ([k, v]) =>
+            `${k}: ${this.serializeInputValue(v, objectFields[k].type)}`,
+        )
         .join(', ');
       return `{${objectItems}}`;
     }
-    if (input instanceof EnumValue) {
-      return input.toString();
-    }
 
     throw new Error(`invalid type for input: ${input}`);
   }
 
-  serializeArgs(args: Obj) {
+  serializeArgsInline(args: Obj, argDefinitions: g.GraphQLArgument[]) {
+    const argsByName = keyBy(argDefinitions, (a) => a.name);
     const serializedArgs = Object.entries(args)
       .filter(([, v]) => v !== undefined)
-      .map(([k, v]) => `${k}: ${this.serializeInputValue(v)}`)
+      .map(
+        ([k, v]) => `${k}: ${this.serializeInputValue(v, argsByName[k].type)}`,
+      )
       .join(', ');
 
     return serializedArgs && `(${serializedArgs})`;
   }
 
+  serializeVariableDefinitions(
+    argNames: string[],
+    argDefinitions: g.GraphQLArgument[],
+  ): [string, string] {
+    if (argNames.length === 0) {
```

Comment:
Might want to put a comment here as to why you can't just return `['()', '()']` -- I tested it and graphql does complain, but it's not obvious (IMO at least) that `mutation () {` wouldn't be allowed

Reply:
or alternatively just doing a last step map to transform any value that is empty to empty string? probably what you have is cleanest, just throwing out some ideas